### ISSUE TITLE
Eye Dome Lighting broken in VR mode

### ISSUE THREAD

**Issue by lilleyse** [CesiumGS/cesium#8704]
[Sandcastle](https://sandcastle.cesium.com/#c=zVltc9u4Ef4raPpB8oxMkSIpUbLja2K7vcwkTSZSctOe7wNEwhIakuABoGT1Jv+9C/BdpGw5jS/WjMciuYvdffDsC6gN5mhDyZZw9BLFZIsuiaBpZHzW9/o9X19eslhiGhPeG6A/bmIEH0k4h1sfONvQgPBZoehzgiX5hfEwWGQi/ZNBprLhr1MpWTxDkqfkJv56cnYT38QbcEH4JCbgQeaKoS/Pskfq1jsWkHBBQyKI1Dr0FvX/klv8wGgsL0OWBvM1Dmi8MqiYp0nCuCRBXy91clK4vaVxwLYGDgmX/d5iTQVacrYVEH/AiEAxk0hkyihRCyNfrYxEtnRPufxVeXCbxr6kLEZcOdUvDdQjMBJOIyrphgiDk4htSH8/GrVeoVa/D1ikcUBuAfSgMDkcosWaVKIAI/a/CCThppCAOmK3iKUcRTSmCCdJSH2sfDQyJJMSKAHL/9qbSwP9TEISC9jW3uU65f6699s+7CCah0bucJSEZLFLAKhZfbl8g/2UcxLL60quKfar+VsuGeE7GqXRHNhC4nmCfXLNOeMgbo0NMxdaERYRyamvn8192DQlUD7PF3klJYlTHSg8NgcIcLr+PaUbkIf9k6xCMtNbYkE+EsHC9HgdsiNX4M1bulpL4MFcQqAruW74syfzERiTikwCdvCswRqZbfOCfS5w7ue3Gkzao4SspYAWaexqTv9Kzmg922ebcWgjqjUOSLRW6tqtl23vuuQOeVXf2a6lOsR+Ok4M9rxldI8WXQb3RH56WKTL0CEudVk8IPvQmjn3jlgxk2xVtZDhYC6z6lAVN8iSORBRFRwoMZLiEPk4IhxrT1TahIx9QViid1BbqvpSsTVX+8AEzVHOq/glhnItKI5t45azSLmFwfTpyLBsx/EmtuXZE890PGeATMMzJ5btmaPJxJ6YkwnccyeObdhTx5lOpt7IGo/KylpZfc8pZHfBp1q3+5lobD5Q6a8/sjDULlyRFVBe9C1rZEyn7nQ8nlgjy7Rdb4BOR5ZhO/bUdF3Tm4ycsaW8AqfGU9eaWp439RzLrRf3sidkgBmQWir1+zmy6hMQATui3ZvtAzWoxFgVxawjspokiYMFBxRvGY/KBv0OQ/LdOcabq+t/Lt4s/pWJ57043+Tm7iH7ChU1aAUhcGg1AVIQobev5ijJRoAALXdoLWUiZsPhdrs1QroMsTAYXw2FbghiWBrQjTevMCgV0E84SUKoLZEqv1xXXv0VxwFaY1H1AkR0fYLuxhmkNbgS7hD5PQUqSlaurxiKN+DpSjEVZpWslwcU+mTsq3uIYH+tXTAqnsiy0tbIkf2zr3II+n+glIclnG9YrNIdeifRrHklQOZN0HeBoCcZrPeNBTgIysJf7sADpRfcU13yrCl8VN2rTxUKp19oGKIl0S0xQCBwyCSNhYQkechkq4bWh5j7Vbv7h3VEnLgRoJotH1TZK4KVWkOxqzt3VspsdmrUST2VIm1UJQbMybcG+kCAiuhVCFk1QJeAUrSkGH2KgQoc8nyHPrIlk9QX6C1eQgr6OI6Ve7DAq5D8B5KBM7TgzPdZSHVuQDYTii4p8zEPacHlzgwDqmnSHZNeJR9UIi0Z7KJ2g90hQf9LEBSUp84fa+yMrOeYQA4oKoCvWNwDAMOQbbMuK1CI+Qr2V65xjByU0DsSCuPRKQN9xNUWAIo0gjoWq5K/gs6gum0iU662sFSAmd9FgIqkkWIXnGeI3EK4sAXA7iWows5lDhrfloOmkfujThRAJGjzwNJ1uyYDmTRfIKKah8afn7/f0nDrjC1nkb7jmNbEcYyx41iOa1swaYxGrulahmOZE29qWaY7GiDHnbrOyDTsiTPxHMe2Tw417Humjr4LZsYTGwYb17SnE3cCg4ZpjCbjMUwWru3ZU88bD9DYGHkWGPYmrmOapueetHr4YwqYvyb+l38Tzvrg22ccpqQ6gkBOFzf/DoVOZXaCucguKvl8vzgBcsaov6fyUnP6BMblshugWXPdwqnm2HmWn3ovWQwFMps7q3NpRKKl4jtwg6EvMfO/sFQitoSz/AYvIVggXo508dTQJ+bqEF5aeE1z7tYO10zfuHr/DoE7qloKlWvq3qc3KsclZCN0T1ULqcyP2JKxcIlVgQmYnyolyCl5nem/3kFd6+Ui+k3CvnvqzL5TvgAtROXmoFg383dfCyy8L6Oua/XaB/LeiSHSpfA5XZJ+QYH2xucvNfJ9VW9bChG9nc0zfamlPq0d1MwECKEm3LuO1V6naKvFKuUro+MRONABGjBkqx8Go4Agt5pTuP0qp6HQdYK/px/dk1V56Pm/xwLQUdJ/UPDHtpsng6Ldzp8NEp2TRkdh/i44NCeOZ4NBaxB6qvgPvFN5NkDc837oqVKj86XQcwWkern1ODjmMKD6WEg4ssCpYcFWq5Bkv0j0e9exwgTVa8NAz5SDKmbNRxKUETw63CNCxe0KUJ6dq/53RCTXO5hdADRUoPZj4mkP6odiejF4cS7kLiQX2ZO/0Uj/CgOHxb5hDCWBEQYOu2K4TEFfGr4QxWb/tRi7ar4tYdBbcXVynSG+WuK+o8b0/M80vDpREnU0jlcz5CR3tdtwdgoIP+WabvWHX/fM0jiBybMODEyrFNrZKQ7pCob+iAZB+Zq7ZvFUsmSGRg2rxSM4YkkW1Z/umzXWcIAgjahvWSxPt+rgJ2fgf1iD+XxYoHse0A2iwcubF3u/7t28QH6IhYAnt2kYzuGof/Pi4nwI8g01NZuBi+8hzBDvlMjaunib3TQM43wIl22t3G0Qz3w6FzAk+hIFWOLTJQy9IMMSxUwAu/5b0wDOIZDas47fl7R72TrFqlLR/+JcLlmwK+/xiwqjcxlcvMv6LcomMaRHMaSnkPMhPG4KN+vVebbbEqyDwxzHK/BC/eQGV3DMUd/xHXwfO/pCSJLoJxZc1EPNYzowEuYxf0pABaR62mivxK7bGUnupDIJ+wZX7qMMNpaugQBfeQ1GwKPkSEY/UKxVTa25p9NE8x/lW4Ns9tVj33dDfXQU6B2D59MB3mHsSLC7ObuP9veAzW7gZumLw9ypefDkPK3Z+ibUXquXUdVY+d0Qs8wm08wDVGtOtU8HV9PO/5PNrcnh4ZxuqaBiaH0yvA/AfWB2fjrcDxj8Jq62YcxG3R8MYubEnwZhZu4YAIdZk4f/uuerGScbOv4H) (make sure to click the VR button)

![bad2](https://user-images.githubusercontent.com/915398/77716697-28a6b880-6fb5-11ea-8c66-d621fb7de262.jpg)

**Comment by emackey** [comment]
Reminder: Cesium's "VR" mode pre-dates actual WebVR, which itself has been folded into WebXR.  Rather than fixing bugs in Cesium VR, consider replacing it with a more modern WebXR integration.

**Comment by mramato** [comment]
@emackey PRs welcome :smile: 

Full support for WebXR is definitely on our long term roadmap, but with so many exciting things happening in the Cesium world, it will most likely be some time before we get to it.

**Comment by emackey** [comment]
Yeah, I might have a tough time justifying to AGI, but it would be a ton of fun, and I do have hardware to properly test it.

Actually my biggest concern about it is that Cesium likes to hand-craft its model and projection matrices for the camera.  In WebXR (and WebVR before it), the model and projection matrices for left and right eyes are handed over by the browser API to the application, using a world space that corresponds to the user's actual environment (their living room or wherever they stand or sit).

So basically we need to short-circuit Cesium's camera logic for figuring these matrices out, and inject the browser-supplied matrices instead, dealing with any downstream dependencies of those appropriately.

After that we would need some UI for allowing the user to arbitrarily scale scene space relative to user-world-space, and change locations etc.  But just getting an in-place WebXR immersive camera in place would make a great first PR on this front.

I don't think this is the kind of project I could embark on alone. I would want a collaborator with some understanding of Cesium camera/view internals.

**Comment by mramato** [comment]
While it's probably out of date, we do have an issue for this: #3422  Please feel free to provide any and all thoughts there so that when we go to do it we have the research all in one place.  Thanks!

**Comment by Winley113** [comment]
How can I get the full source code of this API
